examples/anvil/examples/deploy_contract_anvil.rs

```diff
@@ -39,33 +35,28 @@ async fn main() -> Result<()> {
 
     // Create a provider with a signer and the network.
     let rpc_url = anvil.endpoint().parse()?;
-    let provider =
-        ProviderBuilder::new().signer(EthereumSigner::from(signer)).on_reqwest_http(rpc_url)?;
+    let provider = ProviderBuilder::new()
+        .with_recommended_fillers()
+        .signer(EthereumSigner::from(signer))
+        .on_http(rpc_url)?;
 
     println!("Anvil running at `{}`", anvil.endpoint());
 
-    // Get the base fee for the block.
-    let base_fee = provider.get_gas_price().await?;
-
     // Deploy the contract.
     let contract_builder = Counter::deploy_builder(&provider);
-    let estimate = contract_builder.estimate_gas().await?;
-    let contract_address =
-        contract_builder.gas(estimate).gas_price(base_fee).nonce(0).deploy().await?;
-
-    println!("Deployed contract at address: {contract_address:?}");
+    let contract_address = contract_builder.deploy().await?;
+    let contract = Counter::new(contract_address, provider);
```

Comment:
Alternatively to be replaced by

```rust
Counter::deploy(&provider).await?;
```

@prestwich this example, as is, throws the following error: `missing properties: [("NonceManager", ["from"]), ("Signer", ["TODO"])]` which is unexpected

Am I missing something or is this a bug?

Reply:
not a bug but work blocked by a PR. See the comment on the relevant code in the signer filler

Reply:
more specifically, Signer setting From and reporting kissing keys is blocked by refactor of tx builder